030-Concepts/075-file-attachments.mdx

```diff
@@ -187,6 +187,67 @@ user = xata.records().update("Users", "record_id", {
 img = xata.files().transform("https://us-east-1.storage.xata.sh/4u1fh2o6p10blbutjnphcste94", {
   "quality": 50
 })
-````
+```
+
+</TabbedCode>
+
+### Upload URLs
+
+An upload URL provides a secure and authenticated method to upload or update a file without requiring an API key. This URL includes a time-limited signature, that ensures that only users with the URL can upload a file within a set time frame.
+
+The URLs are similar to signed URLs in that they must be specifically requested and are not automatically included with a standard record request. This improves security by controlling access and limiting the window during which uploads can occur.
+
+HTTP `PUT` is the only valid method for a request using the upload URL.
+
+An upload URL is assigned to a specific file. Subsequent uploads using the same URL will overwrite the existing file content, instead of generating additional files. Every time the URL is used, it updates the same specific file it's linked to.
+
+<TabbedCode tabs={['TypeScript', 'Python']}>
+
+```ts
+// Returns the upload URL for records[0].photo.uploadUrl
+const records = await xata.db.Users.select(['id', 'name', 'photo.*', 'photo.uploadUrl']).getMany();
+
+// Returns an empty string for records[0].photo.uploadUrl
+const records = await xata.db.Users.getMany();
+```
+
+```python
+# Returns the signed URL for records[0].photo.uploadUrl
+records = xata.data().query("Users", {
+  "columns": ["id", "name", 'photo.*', 'photo.uploadUrl']
+})
+
+# Returns an empty string for records[0].photo.uploadUrl
+records = xata.data().query("Users")
+```
 
 </TabbedCode>
+
+Upload URLs have a configurable time to live (TTL), which specifies when access to the URL expires. The default is 24h. You can modify the timeout duration of the upload URL for each file by adjusting the `uploadUrlTimeout` field within the file object. The value represents a duration in seconds.
+
+Use upload URLs when you need temporary write access without revealing the API key, such as exposing uploads in a web application.
+
+<TabbedCode tabs={['TypeScript', 'Python']}>
```

Comment:
Should these examples include Go?

Reply:
Yes, they should. Will take care of that.